### PR TITLE
@W-14689505@: Action Implementation (Part 1 of 2) - all except results analysis

### PR DESCRIPTION
**Pull request by stephen-carter-at-sf**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "$schema": "https://json.schemastore.org/tsconfig",
+  "compilerOptions": {
+    "rootDir": ".",
+    "target": "ES2022",
+    "module": "NodeNext",
+    "moduleResolution": "NodeNext",
+    "baseUrl": "./",
+    "sourceMap": true,
+    "outDir": "./dist_local",
```
Comment (stephen-carter-at-sf) [inline tsconfig.json:10]: Note that the npm run package command uses ncc to compile the index.ts and all its dependencies into a single index.js file inside of the dist folder. In fact there is a github action to confirm that we don't accidentally submit anything else from dist. To make things neat and organized for local development, i have chosen to have the compile to compile to dist_local instead which is on the .gitignore list now. Also note that this ncc really helps us not have to worry about accidentally shipping the tests - so I was able to make the root directory "." and include the tests in the main build and linting.

File: src/main.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ * The main function for the action.
+ * @returns {Promise<void>} Resolves when the action is complete.
+ */
+export async function run(dependencies: Dependencies): Promise<void> {
+    try {
+        dependencies.startGroup('Preparing Environment')
+        // TODO: NICE TO HAVES:
+        // * Verify that Salesforce CLI "sf" is installed
+        // * Verify that sfdx-scanner plugin is installed (and if not, then install it as a separate step)
+        // * Echo version of sfdx-scanner in use
```
Comment (stephen-carter-at-sf) [inline src/main.ts:21]: I hope to do these along with the output analysis in the next work item. And I might break it up into 2 separate PRs to make things easy.

File: __tests__/utils.test.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        expect(outfile2).toEqual('some==other=file.json')
+    })
+
+    it('Test quote in file name', async () => {
+        const outfile1 = extractOutfileFromRunArguments('--outfile "some\'file.json"')
+        expect(outfile1).toEqual("some'file.json")
+    })
+
+    it('Test trailing -o', async () => {
+        const outfile1 = extractOutfileFromRunArguments('--normalize-severity -o  ')
```
Comment (jfeingold35) [inline __tests__/utils.test.ts:70]: When this command actually tries to run, they'll get an error because the flag expects a value.

**Review by jfeingold35 — APPROVED** [review]

File: __tests__/utils.test.ts (hunk 3 above)
Comment (stephen-carter-at-sf) [reply __tests__/utils.test.ts:70]: I understand. This is just an extra added level of protection.

**Pull request merged**